test/e2e/create_test.go

```diff
@@ -149,6 +218,24 @@ func (tc *testContext) waitForWindowsMachines(machineCount int, phase string) er
 			if machine.Status.Phase == nil || *machine.Status.Phase != phase {
 				return false, nil
 			}
+			// If waiting for a provisioned Machine, ensure the Machine is fully attached to the network and has an
+			// assigned IPv4 address.
+			if phase == "Provisioned" {
```

Comment:
I suggest a negation check to reduce the indentation.
```suggestion
                      if phase != "Provisioned" {
                           return...
}
```